src/BlockTimer.svelte

```diff
@@ -0,0 +1,34 @@
+<script lang="ts">
+  import type { Config } from "@app/config";
+
+  export let config: Config;
+  export let startBlock: number;
+  export let duration: number;
+
+  let currentBlock: number = startBlock;
+
+  config.provider.on("block", (latestBlock: number) => {
+    if (startBlock + duration > currentBlock) currentBlock = latestBlock;
+  });
+</script>
+
+<style>
+  #parent {
+    text-align: center;
+    height: 20px;
+    width: 290px;
+    border-radius: 5px;
+    background-color: var(--color-secondary-background);
+  }
+  #loader {
+    color: #fff;
+    height: 20px;
+    width: 0px;
+    border-radius: 5px;
+    background-color: var(--color-secondary);
+  }
```

Comment:
We generally use classes instead of ids for our components, if nothing but for consitency.